Tools/refcounts/lint.py

```diff
@@ -0,0 +1,301 @@
+
+    def param(self, sig: Signature, param: Param, message: str) -> None:
+        self.count += 1
+        fullname = f'{sig.name}[{param.name}]'
+        print(f'{flno_(param.lineno)} {fullname:50} {message}')
+
+def check(view: FileView) -> None:
+    w = CheckerWarnings()
+
+    for sig in view.signatures.values():  # type: Signature
+        # check the return value
+        rparam = sig.rparam
+        if not rparam.ctype:
+            w.block(sig, 'missing return value type')
+        if rparam.reftype is RefType.UNKNOWN:
+            w.block(sig, 'unknown return value type')
+        # check the parameters
+        for name, param in sig.params.items():  # type: (str, Param)
+            ctype, reftype = param.ctype, param.reftype
+            if ctype in OBJECT_TYPES and reftype is RefType.UNUSED:
+                w.param(sig, param, 'missing reference count management')
+            if ctype not in OBJECT_TYPES and reftype is not RefType.UNUSED:
+                w.param(sig, param, 'unused reference count management')
+            if name != C_ELLIPSIS and not name.isidentifier():
+                # Python accepts the same identifiers as in C
+                w.param(sig, param, 'invalid parameter name')
+
+    if w.count:
+        print()
+        print(f'Found {w.count} issue(s)')
+    names = view.signatures.keys()
+    if sorted(names) != list(names):
+        print('Entries are not sorted')
+
+def check_structure(view: FileView, stable_abi_file: str) -> None:
+    print(f"Stable ABI file: {stable_abi_file}")
+    print()
+    stable_abi_str = Path(stable_abi_file).read_text()
+    stable_abi = tomllib.loads(stable_abi_str)
+    expect = stable_abi['function'].keys()
+    # check if there are missing entries (those marked as "TODO" are ignored)
+    actual = IGNORE_LIST | view.incomplete | view.signatures.keys()
+    if missing := (expect - actual):
+        print(f'Missing {len(missing)} stable ABI entries:')
+        for name in sorted(missing):
+            print(name)
+
+STABLE_ABI_FILE_SENTINEL = object()
+
+def _create_parser() -> ArgumentParser:
+    parser = ArgumentParser(
+        prog='lint.py',
+        formatter_class=RawDescriptionHelpFormatter,
+        description='Lint the refcounts.dat file.\n\n'
+                    'Use --abi or --abi=FILE to check against the stable ABI.',
+    )
+    parser.add_argument('file', nargs='?', default=DEFAULT_REFCOUNT_DAT_PATH,
+                        help='the refcounts.dat file to check '
+                             '(default: %(default)s)')
+    parser.add_argument('--abi', nargs='?', default=STABLE_ABI_FILE_SENTINEL,
+                        help='check against the given stable_abi.toml file '
+                             '(default: %s)' % DEFAULT_STABLE_ABI_TOML_PATH)
+    return parser
+
+def main() -> None:
+    parser = _create_parser()
+    args = parser.parse_args()
+    lines = Path(args.file).read_text().splitlines()
```

Comment:
PEP 686 doesn't go in till 3.15 :smile:
```suggestion
    lines = Path(args.file).read_text(encoding="utf-8").splitlines()
```